utils/DeployUtils.ps1

```diff
@@ -259,24 +287,111 @@ function ConfigureScubaGearModule {
+        # Write-Warning $_.ScriptStackTrace
+        Write-Warning ">>> Exception:"
         Write-Warning $_.Exception
-        Write-Error "Failed to test module manifest"
-        return $False
+        Write-Error ">>> Failed to test module manifest."
     }
 
     # True indicates that the updating and testing were successful.
     return $True
 }
 
-function CreateFileList {
+function SignScubaGearModule {
     <#
-    .NOTES
-    Internal function
+        .SYNOPSIS
+            Code sign the specified module
+        .Description
+            This function individually signs PowerShell artifacts (i.e., *.ps1, *.pms1) and creates a signed catalog of the entire module using a certificate housed in an Azure key vault.
+        .Parameter AzureKeyVaultUrl
+            The URL of the key vault with the code signing certificate
+        .Parameter CertificateName
+            The name of the code signing certificate
+        .Parameter ModulePath
+            The root path of the module to be signed
+        .Parameter TimeStampServer
+            Time server to use to timestamp the artifacts
+        .NOTES
+            There appears to be limited or at least difficult to find documentation on how to properly sign a PowerShell module to be published to PSGallery. This page shows general guidance:
+            https://learn.microsoft.com/en-us/powershell/gallery/concepts/publishing-guidelines?view=powershellget-3.x
+            There is anecdotal evidence to sign all PowerShell artifacts (ps1, psm1, and pdsd1) in additional to a signed catalog. For example, Microsoft.PowerApps.PowerShell (v1.0.34) and see both *.psd1 and *.psm1 files are signed and a catalog provided. There are a number of non-authoritive references such as below showing all ps1, psm1, and psd1 being signed first then cataloged:
+            https://github.com/dell/OpenManage-PowerShell-Modules/blob/main/Sign-Module.ps1
     #>
+    param (
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({ [uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'https' })]
+        [System.Uri]
+        $AzureKeyVaultUrl,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $CertificateName,
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({ Test-Path -Path $_ -PathType Container })]
+        $ModulePath,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({ [uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'http', 'https' })]
+        $TimeStampServer = 'http://timestamp.digicert.com'
+    )
+
+    Write-Warning ">> Signing ScubaGear module..."
+
+    # Sign scripts, manifest, and modules
+    $ArrayOfFilePaths = CreateArrayOfFilePaths `
+        -SourcePath $ModulePath `
+        -Extensions "*.ps1", "*.psm1", "*.psd1"  # Array of extensions
+    if ($ArrayOfFilePaths.Length -eq 0)
+    {
+        Write-Error "Failed to find any .ps1, .psm1, or .psd files."
```

Comment:
Should be .psd1 files, or should search for .psd files in the array of extensions

Reply:
Fixed.